resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ExecutorRunnableSuite.scala

```diff
@@ -103,4 +106,33 @@ class ExecutorRunnableSuite extends SparkFunSuite {
     val metadataStorageVal: Any = metaInfo.get(SHUFFLE_SERVER_RECOVERY_DISABLED.key)
     assert(metadataStorageVal != null && metadataStorageVal.asInstanceOf[Boolean])
   }
+
+  test("SPARK-53209: ActiveProcessorCount should default to 1 when executor cores not configured") {
+    val sparkConf = new SparkConf()
+    val execRunnable = createExecutorRunnable(sparkConf)
+
+    val commands = execRunnable.prepareCommand()
+    commands should contain ("-XX:ActiveProcessorCount=1")
+    commands should contain inOrderElementsOf List("--cores", "1")
+  }
+
+  test("SPARK-53209: ActiveProcessorCount should match configured executor cores") {
+    val sparkConf = new SparkConf()
+      .set(EXECUTOR_CORES, 4)
+    val execRunnable = createExecutorRunnable(sparkConf)
+
+    val commands = execRunnable.prepareCommand()
+    commands should contain ("-XX:ActiveProcessorCount=4")
+    commands should contain inOrderElementsOf List("--cores", "4")
+  }
+
+  test("SPARK-53209: ActiveProcessorCount should respect custom executor core count") {
+    val sparkConf = new SparkConf()
+      .set(EXECUTOR_CORES, 7)
+    val execRunnable = createExecutorRunnable(sparkConf)
+
+    val commands = execRunnable.prepareCommand()
+    commands should contain ("-XX:ActiveProcessorCount=7")
+    commands should contain inOrderElementsOf List("--cores", "7")
+  }
```

Comment:
Above two tests seem to be identical (except the number of cores)? Is there anything else special we are testing here?